src/components/Filter/useFiltersFromUrl.ts

```diff
@@ -0,0 +1,54 @@
+import { useMemo, useState } from 'react';
+import { useSearchParams } from '_/hooks/useSearchParams';
+
+import { Field } from '../types';
+
+import { GlobalFilters } from './types';
+
+const safeParse = (str: string) => {
+  try {
+    return JSON.parse(str);
+  } catch (e) {
+    return undefined;
+  }
+};
+
+export const useFiltersFromUrl = ({
```

Comment:
A quick doc comment would be nice....

"Init and maintain a set of filters on the search part of the URL"